### PR TITLE
Convert badges to CSS variables

### PR DESCRIPTION
**Pull request by mdo**

### CODE REVIEW

File: scss/_badge.scss [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
 // Requires one of the contextual, color modifier classes for `color` and
 // `background-color`.
 
 .badge {
+  // scss-docs-start badge-css-vars
+  --#{$variable-prefix}badge-padding: #{$badge-padding-y $badge-padding-x};
+  @include rfs($badge-font-size, --#{$variable-prefix}badge-font-size);
+  --#{$variable-prefix}badge-font-weight: #{$badge-font-weight};
+  --#{$variable-prefix}badge-color: #{$badge-color};
+  --#{$variable-prefix}badge-bg: transparent;
```
Comment (Kopyov) [inline scss/_badge.scss:14]: This var is declared, but not used

**Review by ffoodd — APPROVED** [review]
`--bs-badge-bg` should be used indeed, but apart from that it looks good 👌

File: site/content/docs/5.1/components/badge.md [hunk 2]
```diff
@@ -86,8 +86,16 @@ Use the `.rounded-pill` utility class to make badges more rounded with a larger
...
 {{< /example >}}
 
-## Sass
+## CSS
 
 ### Variables
 
+<small class="d-inline-flex px-2 py-1 font-monospace text-muted border rounded-3">Added in v5.3.0</small>
+
+As part of Bootstrap's evolving CSS variables approach, buttons now use local CSS variables on `.badge` for enhanced real-time customization. Values for the CSS variables are set via Sass, so Sass customization is still supported, too.
```
Comment (julien-deramond) [inline site/content/docs/5.1/components/badge.md:95]: `buttons now use [...]` → `badges now use [...]`

File: site/content/docs/5.1/components/badge.md [hunk 3]
```diff
@@ -86,8 +86,16 @@ Use the `.rounded-pill` utility class to make badges more rounded with a larger
 {{< /badge.inline >}}
 {{< /example >}}
 
-## Sass
+## CSS
 
 ### Variables
 
+<small class="d-inline-flex px-2 py-1 font-monospace text-muted border rounded-3">Added in v5.3.0</small>
```
Comment (XhmikosR) [inline site/content/docs/5.1/components/badge.md:93]: FIXME if we land this in 5.2.0

**Comment by ffoodd** [comment]
Regarding background, I wonder if we shouldn't reintroduce colors in badges, using CSS vars to only hydrate `color` and `background-color` (as we do in alerts already). That'd fix #34016 (which is requested several times already) without introducing any BC since existing usages based on utilities would still apply.

If not (or not yet) we may simply drop the useless CSS var and merge it almost as-is.

Any thought, @XhmikosR?

I just noticed a doc sentence to change regarding `.bg-light` and dark text: it also applies to warning and info theme colors. I'll take care of this when possible.

**Comment by mdo** [comment]
Tried getting the `browserstack (push)` to pass a few times, so giving up and merging anyway :).

**Pull request merged**